src/java.base/windows/native/libnio/ch/Net.c

```diff
@@ -539,10 +534,11 @@ Java_sun_nio_ch_Net_joinOrDrop6(JNIEnv *env, jobject this, jboolean join, jobjec
     } else {
         int opt = (join) ? MCAST_JOIN_SOURCE_GROUP : MCAST_LEAVE_SOURCE_GROUP;
         n = setGroupSourceReqOption(env, fdo, opt, group, index, source);
+        error_message = "setsockopt with group source request";
     }
 
     if (n == SOCKET_ERROR) {
-        handleSocketError(env, WSAGetLastError());
+        NET_ThrowNew(env, WSAGetLastError(), error_message);
```

Comment:
I this this one can be simplified to "setsockopt".  Source-specific multicast isn't widely used but if you did happen to get a failure here then the stack trace should make it very clear.